docs/argocd-app-of-apps.md

```diff
@@ -26,7 +26,70 @@ we consider valuable on most clusters. As with all the charts, `infra-apps` can
 depending on your exact situation, for example on Red Hat OpenShift you would typically not deploy the bundled ingress
 component.
 
-## The `argoconfig` library chart
+### Structure of app-of-app charts
+
+The charts all share a similar structure that looks as follows:
+
+```
+.
+├── README.md
+├── Chart.lock
+├── Chart.yaml
+├── values.yaml
+├── ci
+│   └── default-values.yaml
+├── examples
+│   └── example-app.yaml
+└── templates
+    ├── NOTES.txt
+    └── example-app.yaml
+```
+
+In this example the chart contains a single application called `example-app`. Most files correspond to their regular
+helm chart counterparts but there are some special files. App names are dasherized in file names and camel cased
+when used in YAML keys. As an example, the application "Example App" has files names using `example-app` and keys
+using `exampleApp` throughout this documentation.
+
+`ci/defauls-values.yaml` is not specific to app-of-apps charts but it enables each app during chart testing with
+GitHub Actions:
+```yaml
+exampleApp:
+  enabled: true
+  values: {}
+    # values needed for testing defined here
+```
+
+`examples/example-app.yaml` contains an example on how to configure the given app using the chart:
+```yaml
+exampleApp:
+  enabled: true
+  values:
+    ingress:
+      enabled: true
+      annotations:
+        kubernetes.io/ingress.class: infra-ingress
```

Comment:
small correction : `kubernetes.io/ingress.class: nginx`

Reply:
Thanks for catching this one, i shall now prepare a PR to fix this everywhere else as well :stuck_out_tongue_closed_eyes: